### PR TITLE
Feat/depub

### PR DESCRIPTION
**Pull request by lancatlin**
+ Allow user to depub a message by message context menu
+ Respond iscn widget link to let user finish the transaction

Other changes:
+ Refactor the structure of commands
+ Combine register-commands into main workflow

### CODE REVIEW

**Review by WeiJun0827 — APPROVED** [review]

File: src/register-commands.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+} from './config.js';
+
+async function registerCommands() {
+  const commandPath = path.join(__dirname, 'commands');
+  const commandFiles = fs.readdirSync(commandPath).filter((file) => file.endsWith('.js'));
+
+  const commands = await Promise.all(commandFiles.map(async (file) => {
+    const filePath = path.join(commandPath, file);
+    const command = (await import(filePath)).default
+    console.log(command.data.name);
```
Comment (williamchong) [inline src/register-commands.js:22]: Please print more context if this `console.log` is suppose to be kept

File: src/register-commands.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  CLIENT_ID,
+  GUILD_ID,
+  TOKEN,
+} from './config.js';
+
+async function registerCommands() {
+  const commandPath = path.join(__dirname, 'commands');
+  const commandFiles = fs.readdirSync(commandPath).filter((file) => file.endsWith('.js'));
+
+  const commands = await Promise.all(commandFiles.map(async (file) => {
```
Comment (williamchong) [inline src/register-commands.js:19]: By dynamically importing these js from `commands`, does that mean we will lose lint checking/future typescript support?

File: src/register-commands.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+
+  const rest = new REST({ version: '9' }).setToken(TOKEN);
+
+  rest.put(
+    Routes.applicationGuildCommands(CLIENT_ID, GUILD_ID),
+    { body: commands.map((command) => command.data.toJSON()) },
+  )
+    .then(() => console.log('Successfully registered application commands.'))
+    .catch(console.error);
```
Comment (williamchong) [inline src/register-commands.js:34]: Please stick to `await` if we are using `async` in function

File: src/commands/depub.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import { WIDGET_ENDPOINT } from '../config.js';
+
+export default {
+  data: new ContextMenuCommandBuilder()
+    .setName('depub')
+    .setType(ApplicationCommandType.Message),
+  async execute(interaction) {
+    const msg = await interaction.channel.messages.fetch(interaction.targetId);
+    const msgUrl = `https://discord.com/channels/${msg.guildId}/${msg.channelId}/${msg.id}`;
+    const msgContent = msg.content.substring(0, 200);
```
Comment (williamchong) [inline src/commands/depub.js:13]: Please add `...` in the end if we actually substring?
maybe use Unicode Character “…” (U+2026) if thats save some space

File: src/commands/depub.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { ContextMenuCommandBuilder } from '@discordjs/builders';
+import { ApplicationCommandType } from 'discord-api-types/v10';
```
Comment (williamchong) [inline src/commands/depub.js:2]: Are we mixing `v10` here and `v9` api type below?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/commands/depub.js (hunk 4 above)
Comment (lancatlin) [reply src/commands/depub.js:13]: I think unicode whold be encoded to %E2%80%A6 since it's not IRI.

File: src/register-commands.js (hunk 2 above)
Comment (lancatlin) [reply src/register-commands.js]: Yes it affects. I removed the dynamic importing.

File: src/commands/depub.js (hunk 5 above)
Comment (lancatlin) [reply src/commands/depub.js:2]: Updated all to v10

File: src/commands/depub.js [hunk 6]
```diff
@@ -12,14 +12,14 @@ export default {
...
-    const guildName = interaction.member.guild.name
+    const guildName = interaction.member.guild.name;
     iscnUrl.search = new URLSearchParams({
       title: `depub.space-${(new Date()).toISOString()}`,
       url: msgUrl,
       type: 'message',
       tags: `Discord,${guildName}`,
       publisher: 'depub',
-      description: `${msgUrl}\n${msg.author.username}: ${msgContent}\n#Discord #${guildName}`,
+      description: `${msgUrl}\n${msg.author.username}: ${msgContent}...\n#Discord #${guildName}`,
```
Comment (williamchong) [inline src/commands/depub.js:22]: Can we add `...` only when the content actually hit the string length limit?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/commands/depub.js (hunk 6 above)
Comment (lancatlin) [reply src/commands/depub.js]: Updated

**Review by WeiJun0827 — APPROVED** [review]

File: src/commands/depub.js [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+export default {
+  data: new ContextMenuCommandBuilder()
+    .setName('Share to DePub')
+    .setType(ApplicationCommandType.Message),
+  async execute(interaction) {
+    const msg = await interaction.channel.messages.fetch(interaction.targetId);
+    const msgUrl = `https://discord.com/channels/${msg.guildId}/${msg.channelId}/${msg.id}`;
+    const isTooLong = msg.content.length > 200;
+    const msgContent = msg.content.substring(0, 200);
```
Comment (nwingt) [inline src/commands/depub.js:14]: 200 can be a const or config?

File: src/commands/translate.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+export default {
+  data: new SlashCommandBuilder()
+    .setName(COMMAND_NAME)
+    .setDescription('Translate between Cosmos address and Like address')
+    .addStringOption((address) => address.setName(COMMAND_OPTION_NAME)
+      .setDescription('Cosmos address (cosmos1...) or Like address (like1...)')
+      .setRequired(true)),
+  async execute(interaction) {
+    if (interaction.channel.id !== TARGET_CHANNEL_ID) {
```
Comment (nwingt) [inline src/commands/translate.js:16]: TARGET_CHANNEL_ID seems too generic for `/translate`
how about sth like  `CHANNEL_ID_FOR_TRANSLATE`

**Pull request merged**